Eevee.Sleep.Bot/Utils/DiscordMessageMaker.cs

```diff
@@ -171,4 +171,22 @@ public static class DiscordMessageMaker {
             .WithCurrentTimestamp()
             .Build();
     }
+
+    public static MessageComponent MakeRoleSelectButton(ulong[] hasRoles, IEnumerable<SocketRole> guildRoles) {
+        var builder = new ComponentBuilder();
+
+        foreach (var role in hasRoles) {
+            builder.WithButton(
+                label: guildRoles.Where(x => x.Id == role).Select(x => x.Name).First(),
```

Comment:
Make `guildRoles` be a dict of `roleId => name`, then ref this dict to avoid `O(x^2)`.